src/HealthGPS.Core/identifier.cpp

```diff
@@ -55,4 +55,13 @@ std::ostream &operator<<(std::ostream &stream, const Identifier &identifier) {
     stream << identifier.to_string();
     return stream;
 }
-} // namespace hgps::core
\ No newline at end of file
+
+void from_json(const nlohmann::json &j, Identifier &id) { id = Identifier{j.get<std::string>()}; }
+
+void from_json(const nlohmann::json &j, std::map<Identifier, double> &map) {
+    for (auto &[key, value] : j.items()) {
+        map.emplace(Identifier{key}, value.get<double>());
```

Comment:
I *think* you can just construct the `Identifier` in place here:

```suggestion
        map.emplace(key, value.get<double>());
```